impl/src/main/java/org/ehcache/loaderwriter/writebehind/BatchingLocalHeapWriteBehindQueue.java

```diff
@@ -152,9 +153,11 @@ private Batch newBatch() {
     return executor.submit(batch);
   }
   
+  @SuppressFBWarnings("IS2_INCONSISTENT_SYNC")
   @Override
   public long getQueueSize() {
-    return executorQueue.size() * batchSize;
+    return executorQueue.size() * batchSize   // This multiplication with batchSize is still not accurate as the batches are not always full
+           + (openBatch == null ? 0 : openBatch.size());
```

Comment:
As discussed in chat, the openBatch reference could change between the two accesses to it, better create a local var for it.